### PR TITLE
Working through dependency hell

### PR DESCRIPTION
**Pull request by n8fr8**
Build tool updates, moves from jars to dependencies, and NPE/race condition fixes

### CODE REVIEW

File: lib/src/main/java/scal/io/liger/ZipHelper.java [hunk 1]
```diff
@@ -273,7 +273,7 @@ public static InputStream getThumbnailInputStreamForItem(@NonNull ExpansionIndex
         return null;
     }
 
-    private static ArrayList<String> expansionPaths;
```
Comment (vitriolix) [inline lib/src/main/java/scal/io/liger/ZipHelper.java:276]: While fixing a race/NPE is good, the reason this was structured this was as a pretty important perf optimization:

https://github.com/n8fr8/storypath/commit/3815d9812bd7eced5a56afb900dede2485da9763

I'd rather fix the root crash or more carefully consider this refactor with a decent amount of testing to show that this isn't going to significantly impact our performance.

**Comment by vitriolix** [comment]
I don't want to merge e1df762ba707ab08f6118b1fb5ebadf618a20b4d without some discussion and testing.  We did a bunch of that optimization for very good reason, before that the app would lock up a lot, it has some bad exploding n issues when you have a lot of saved stories.  lets pull that into its own PR separate from all these nice build fixes that I'd love to merge

**Comment by n8fr8** [comment]
Happy to discuss. What I saw was one thread trying to use expansionPaths while another was nulling it out. That is all I was trying to solve, as it was causing many crashes from clean install state. I will try to revert it and see how it goes.

**Comment by vitriolix** [comment]
ok, so maybe the correct answer is just using thread syncronization?  if the issue was one thread nulling it after another thread already got past its null check, that seems like the answer?

**Pull request merged**